graylog2-web-interface/src/components/users/TokenList.jsx

```diff
@@ -0,0 +1,137 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import Immutable from 'immutable';
+import Clipboard from 'clipboard';
+
+import { Row, Col, FormControl, ControlLabel, Button, Checkbox } from 'react-bootstrap';
+import UserNotification from 'util/UserNotification';
+import TableList from 'components/common/TableList';
+import TokenListStyle from './TokenList.css';
+
+class TokenList extends React.Component {
+  static propTypes = {
+    tokens: PropTypes.arrayOf(PropTypes.object),
+    delete: PropTypes.func,
+    create: PropTypes.func,
+  };
+
+  static defaultProps = {
+    tokens: [],
+    delete: () => {},
+    create: () => {},
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      token_name: '',
+      hide_tokens: true,
+    };
+
+    this._onNewTokeChanged = this._onNewTokeChanged.bind(this);
+    this._onShowTokensChanged = this._onShowTokensChanged.bind(this);
+    this._createToken = this._createToken.bind(this);
+    this.itemActionsFactory = this.itemActionsFactory.bind(this);
+  }
+
+  componentDidMount() {
+    this.clipboard = new Clipboard('[data-clipboard-button]');
+    this.clipboard.on('success', () => {
+      UserNotification.success('Copied to clipboard!');
+    });
+    this.clipboard.on('error', () => {
+      UserNotification.error('Coping failed!');
+    });
+  }
+
+  componentWillUnmount() {
+    if (this.clipboard) {
+      this.clipboard.destroy();
+    }
+  }
+
+  _onNewTokeChanged(event) {
+    this.setState({ token_name: event.target.value });
+  }
+
+  _onShowTokensChanged(event) {
+    this.setState({ hide_tokens: event.target.checked });
+  }
+
+  _deleteToken(token) {
+    return () => {
+      this.props.delete(token.token);
+    };
+  }
+
+  _createToken() {
+    this.props.create(this.state.token_name);
+    this.setState({ token_name: '' });
+  }
+
+  itemActionsFactory(token) {
+    const copyButtonProps = {};
+    copyButtonProps['data-clipboard-text'] = token.token;
+
+    return (
+      <span>
```

Comment:
I would use a `ButtonToolbar` in here to separate both buttons. If you think putting them together would make more sense, then I would use a `ButtonGroup`, which styles them a bit better for that.